addons/payment_payulatam/controllers/main.py

```diff
@@ -18,5 +21,51 @@ def payulatam_response(self, **post):
         """ PayUlatam."""
         _logger.info('PayU Latam: entering form_feedback with post response data %s', pprint.pformat(post))
         if post:
+            self._verify_signature(post)
             request.env['payment.transaction'].sudo().form_feedback(post, 'payulatam')
         return werkzeug.utils.redirect('/payment/process')
+
+    @http.route('/payment/payulatam/webhook', type='http', auth='public', methods=['POST'], csrf=False)
+    def payulatam_webhook(self, **data):
+        _logger.info("handling confirmation from PayU Latam with data:\n%s", pprint.pformat(data))
+        state_pol = data.get('state_pol')
+        if state_pol == '4':
+            lapTransactionState = 'APPROVED'
+        elif state_pol == '6':
+            lapTransactionState = 'DECLINED'
+        elif state_pol == '5':
+            lapTransactionState = 'EXPIRED'
+        else:
+            lapTransactionState = f'INVALID state_pol {state_pol}'
+
+        data = {
+            'signature': data.get('sign'),
+            'TX_VALUE': data.get('value'),
+            'currency': data.get('currency'),
+            'referenceCode': data.get('reference_sale'),
+            'transactionId': data.get('transaction_id'),
+            'transactionState': data.get('state_pol'),
+            'message': data.get('response_message_pol'),
+            'lapTransactionState': lapTransactionState,
+            'merchantId': data.get('merchant_id'),
+        }
+
+        self._verify_signature(data)
+
+        try:
+            request.env['payment.transaction'].sudo().form_feedback(data, 'payulatam')
+        except ValidationError:
+            _logger.warning(
+                'An error occurred while handling the confirmation from PayU with data:\n%s',
+                pprint.pformat(data))
+        return http.Response(status=200)
+
+    def _verify_signature(self, data):
+        sign = data.get('signature')
+        reference = data.get('referenceCode')
+        tx = request.env['payment.transaction'].sudo().search(
+            [('reference', '=', reference), ('provider', '=', 'payulatam')])
+        sign_check = tx.acquirer_id._payulatam_generate_sign('out', data)
+        if sign_check.upper() != sign.upper():
+            _logger.warning('Invalid signature %s', sign)
+            raise werkzeug.exceptions.Forbidden
```

Comment:
Don't bother extracting the signature check in this PR because:
- The signature will be checked in `_payulatam_form_get_tx_from_data` as soon as you call `form_feedback`.
- You would need to update [line 82 of `payment.py`](https://github.com/odoo/odoo/pull/83923/files#diff-4270f121863b722ad10ee8fc8b4df82c2abdba37896a839fa61839a420ce28abR82) and the translated string just after.
- We don't want to start raising HTTP 403 Forbidden errors in stable.
- `_payulatam_form_get_tx_from_data` already runs the search on the transaction reference.